### PR TITLE
Added Support to read http_proxy enviroment variable

### PR DESCRIPTION
**Pull request by Nirbhay1997**
Signed-off-by: Nirbhay Kumar <nkumar@progress.com>

### Description

Added Support to read http_proxy enviroment variable

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [ ] New functionality includes integration tests/controls
- [ ] New Terraform resources
- [ ] Documentation provided or updated for resources 
- [ ] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: libraries/aws_backend.rb [hunk 1]
```diff
@@ -326,10 +326,12 @@ def initialize(opts)
       client_args[:client_args][:endpoint] = opts[:aws_endpoint] if opts[:aws_endpoint]
       # below allows each resource to optionally and conveniently set max_retries and retry_backoff
       env_hash = ENV.map { |k, v| [k.downcase, v] }.to_h
+      opts[:aws_http_proxy]=   env_hash['aws_http_proxy'] if !opts[:aws_http_proxy] && env_hash['aws_http_proxy']
       opts[:aws_retry_limit]=   env_hash['aws_retry_limit'].to_i if !opts[:aws_retry_limit] && env_hash['aws_retry_limit']
       opts[:aws_retry_backoff]= env_hash['aws_retry_backoff'].to_i if !opts[:aws_retry_backoff] && env_hash['aws_retry_backoff']
       client_args[:client_args][:retry_limit] = opts[:aws_retry_limit] if opts[:aws_retry_limit]
       client_args[:client_args][:retry_backoff] = "lambda { |c| sleep(#{opts[:aws_retry_backoff]}) }" if opts[:aws_retry_backoff]
+      client_args[:client_args][:http_proxy]= opts[:aws_http_proxy] if opts[:aws_http_proxy]
```
Comment (sathish-progress) [inline libraries/aws_backend.rb:334]:
```suggestion
      client_args[:client_args][:http_proxy]= opts[:aws_http_proxy].presence || env_hash['aws_http_proxy'].presence
```

File: libraries/aws_backend.rb [hunk 2]
```diff
@@ -326,10 +326,12 @@ def initialize(opts)
       client_args[:client_args][:endpoint] = opts[:aws_endpoint] if opts[:aws_endpoint]
       # below allows each resource to optionally and conveniently set max_retries and retry_backoff
       env_hash = ENV.map { |k, v| [k.downcase, v] }.to_h
+      opts[:aws_http_proxy]=   env_hash['aws_http_proxy'] if !opts[:aws_http_proxy] && env_hash['aws_http_proxy']
```
Comment (sathish-progress) [inline libraries/aws_backend.rb:329]: this is not needed!

File: libraries/aws_backend.rb [hunk 3]
```diff
@@ -335,6 +335,7 @@ def initialize(opts)
       opts[:aws_retry_backoff]= env_hash['aws_retry_backoff'].to_i if !opts[:aws_retry_backoff] && env_hash['aws_retry_backoff']
       client_args[:client_args][:retry_limit] = opts[:aws_retry_limit] if opts[:aws_retry_limit]
       client_args[:client_args][:retry_backoff] = "lambda { |c| sleep(#{opts[:aws_retry_backoff]}) }" if opts[:aws_retry_backoff]
+      client_args[:client_args][:http_proxy]= opts[:aws_http_proxy].presence || env_hash['aws_http_proxy'].presence
```
Comment (sathish-progress) [inline libraries/aws_backend.rb:338]:
```suggestion
      opts[:aws_http_proxy] || env_hash['aws_http_proxy']
```

**Review by sathish-progress — APPROVED** [review]
LGTM

**Comment by sathish-progress** [comment]
BTW, please add about this variable in readme please @Nirbhay1997

**Comment by Nirbhay1997** [comment]
> BTW, please add about this variable in readme please @Nirbhay1997

Added @sathish-progress , Thanks for the review.

File: README.md [hunk 4]
```diff
@@ -58,6 +58,14 @@ Example `~/.aws/config` :
      # Example configuration
      export AWS_PROFILE="engineering"
  ```
+#### 3) Proxy
+
+Set your http_proxy in a `.envrc` file or export it in your shell. (See example [.envrc file](.envrc_example))
+
+ ```bash
+   export http_proxy="http://proxy.com:123"
```
Comment (sathish-progress) [inline README.md:66]: in code you are referring to this config as `aws_http_proxy`.. I guess this is a typo here than!

**Comment by Nirbhay1997** [comment]
Closing as we already support Custom endpoints.

**Pull request closed without merging**